965.py

```diff
@@ -0,0 +1,27 @@
+# Definition for a binary tree node.
+# class TreeNode:
+#     def __init__(self, val=0, left=None, right=None):
+#         self.val = val
+#         self.left = left
+#         self.right = right
+class Solution:
+    def isUnivalTree(self, root: TreeNode) -> bool:
+        if not root:
+            return True
+
+        checker = set([root.val])
```

Comment:
This set is not needed. You can just compare each child's value with its parent's value whenever you see a child node.

Reply:
Yes, I guess I was sleepy.